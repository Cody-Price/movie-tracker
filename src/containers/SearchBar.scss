.search-bar-div {
  padding-left: 60px;
  height: 50px;
  margin: 175px auto 0 auto;
}

.search-input {
  width: 350px;
  margin-right: 30px;
  z-index: 20;
}

.sticky {
  opacity: 0;
  transition-duration: 0.5s;
}