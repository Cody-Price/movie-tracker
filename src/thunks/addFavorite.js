import { isLoading, addFavorite, hasErrored } from '../actions';

export const postFavorite = (movie, uid) => {
  debugger;
  console.log(uid)
  const url = 'http://localhost:3000/api/users/favorites/new';
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({...movie, user_id: uid})
    })
    if (!response.ok) {
      throw Error(response.statusText)
    }
    dispatch(isLoading(false))
    const result = await response.json();
    dispatch(addFavorite(movie));
  } catch (err) {
    dispatch(hasErrored(err.message))
    }
  }
  
}