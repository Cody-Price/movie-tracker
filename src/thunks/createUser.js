import { isLoading, hasErrored, loginUserAction } from '../actions'
import { loginUser } from './loginUser';

export const createUser = (user) => {
  const url = 'http://localhost:3000/api/users/new'
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify(user)
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const newUser = await response.json();
      const newUserToBeLoggedIn = {
        id: newUser.id,
        email: user.email,
        password: user.password,
        name: user.name
      }
      dispatch(loginUserAction(newUserToBeLoggedIn))
    } catch(err) {
      dispatch(hasErrored(err.message));
    }
  }
}