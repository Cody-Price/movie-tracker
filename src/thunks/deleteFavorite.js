// import { isLoading, deleteFavorite, hasErrored } from '../actions';

// export const deleteFavorite = (movie, uid) => {
//   const url = 'http://localhost:3000/api/users/favorites/new';
//   return async (dispatch) => {
//     try {
//       dispatch(isLoading(true))
//       const response = await fetch(url, {
//       method: 'POST',
//       headers: {
//         'Content-Type': 'application/json'
//       },
//       body: JSON.stringify({...movie, user_id: uid})
//     })
//     if (!response.ok) {
//       throw Error(response.statusText)
//     }
//     dispatch(isLoading(false))
//     const result = await response.json();
//     dispatch(addFavorite(result));
//   } catch (err) {
//     dispatch(hasErrored(err.message))
//     }
//   }
  
// }