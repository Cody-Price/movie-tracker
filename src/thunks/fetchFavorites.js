import { isLoading, addFavorites, hasErrored } from '../actions';

export const fetchFavorites = (userId) => {
  const url = `https://movie-tracker-cody-be.herokuapp.com/api/users/${userId}/favorites`;
  return async (dispatch) => {
    try {
    dispatch(isLoading(true))
    const response = await fetch(url)
    if (!response.ok) {
      throw Error(response.statusText)
    }
    dispatch(isLoading(false))
    const result = await response.json()
    const movies = result.data
    dispatch(addFavorites(movies));
  } catch(err) {
    dispatch(hasErrored(err.message))
    }
  }
}