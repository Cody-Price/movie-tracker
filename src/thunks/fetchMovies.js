import { isLoading, fetchMoviesSuccess } from '../actions'
import cleanMovies from '../API';

export const fetchMovies = async (page) => {
  let selectedPage = page || 1;
  let url = `https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=true&page=${selectedPage}`
  return async (dispatch) =>  {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      dispatch(isLoading(false))
      const result = await response.json()
      const movies = cleanMovies(result.results)
      dispatch(fetchMoviesSuccess(movies))
    } catch (error) {
      // dispatch(hasErrored(true))
      console.log(err.message)
    }
  }
}