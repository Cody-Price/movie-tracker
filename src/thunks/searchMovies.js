import { isLoading, searchMovieSuccess, hasErrored } from '../actions'
import { cleanMovies, SQLsearchString } from '../helper'
import apiKey from '../apiKey'


export const searchMovies = (searchString) => {
  const querySearchString = SQLsearchString(searchString)
  const url = `https://api.themoviedb.org/3/search/movie?api_key=${apiKey}&language=en-US&query=${querySearchString}`
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      const searchedMovies = cleanMovies(result.results)
      dispatch(searchMovieSuccess(searchedMovies))
    } catch (err) {
      dispatch(hasErrored(err.message))
    }
  }  
}
