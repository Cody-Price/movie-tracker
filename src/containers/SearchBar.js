import React, { Component } from 'react';
import './SearchBar.scss';
import { searchMovies } from '../API';
import { fetchMovies } from '../thunks/fetchMovies';
import { connect } from 'react-redux';


class SearchBar extends Component {
  constructor() {
    super()
    this.state = {
      search: ''
    }
  }

  handleChange = (e) => {
    const {name, value} = e.target
    this.setState({
      [name]: value
    })
  }

  handleSubmit = async (e) => {
    e.preventDefault()
    const foundMovies = await searchMovies(this.state.search)
    // this.props.fetchMovies(foundMovies);
    this.resetState()
  }

  resetState = () => {
    this.setState({search: ''})
  }

  render() {
    return (
      <div className="search-bar-div">
        <form onSubmit={this.handleSubmit}>
          <input
            type="text"
            placeholder="Search the Database by any keyword!"
            name="search"
            className="search-input"
            onChange={this.handleChange}
          />
          <button>Search!</button>
        </form>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(fetchMovies(movies))
})

export default connect(null, mapDispatchToProps)(SearchBar);