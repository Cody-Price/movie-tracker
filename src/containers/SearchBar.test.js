import React from 'react'
import { shallow } from 'enzyme'
import { SearchBar, mapDispatchToProps } from './SearchBar'
import { searchMovies } from '../thunks/searchMovies'

jest.mock('../thunks/searchMovies')

describe('search bar', () => {
	let wrapper
	let mockSearchMovies
	let mockSearch
	let mockMovies

	beforeEach(() => {
		mockMovies = [{ title: 'movie1', movie_id: 1}, {title: 'movie2', movie_id: 2}, {title: 'movie3', movie_id: 3 }]
		mockSearchMovies = jest.fn()

		wrapper = shallow( 
			<SearchBar 
				searchMovies={mockSearchMovies}
			/> )
		mockSearch = 'harry potter'
	})

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it.skip('should call handle submit on button click', () => {
		const spy = jest.spyOn(wrapper.instance(), 'handleSubmit')
		wrapper.find('.search-btn').simulate('click')
		expect(spy).toHaveBeenCalled()
	})

	it('should set state of search when search value is typed in', () => {
		const mockEvent = {
			target: {
				name: 'search',
				value: 'harry potter'
			}
		}
		const handleChangeSpy = jest.spyOn(wrapper.instance(), 'handleChange')
		wrapper.instance().forceUpdate()
		const input = wrapper.find('input').at(0)
		input.simulate('change', mockEvent)
		expect(handleChangeSpy).toHaveBeenCalled()
		expect(wrapper.state().search).toEqual('harry potter')
	})

	it.skip('should reset state after submit button is clicked', ()=> {
		const spy = jest.spyOn(wrapper.instance(), 'resetState')
		wrapper.find('.search-btn').simulate('click')
		expect(spy).toHaveBeenCalled()
	})
})

describe('map dispatch to props', ()=> {
	let mockDispatch
	let mockSearch

	beforeEach(() => {
		mockDispatch = jest.fn()
		mockSearch = 'harry potter'
	})
	it('should call dispatch with the correct parameters for search movies', () => {
		const actionToDispatch = searchMovies(mockSearch)
		const result = mapDispatchToProps(mockDispatch)
		result.searchMovies(mockSearch)
		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
	})
})
