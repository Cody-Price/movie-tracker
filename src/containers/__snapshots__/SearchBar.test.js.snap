// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search bar should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    searchMovies={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <input
          className="search-input"
          name="search"
          onChange={[Function]}
          placeholder="Search the Database by any keyword!"
          type="text"
          value=""
        />
        <button
          className="search-btn"
        >
          Search!
        </button>
      </form>,
      "className": "search-bar-div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="search-input"
            name="search"
            onChange={[Function]}
            placeholder="Search the Database by any keyword!"
            type="text"
            value=""
          />,
          <button
            className="search-btn"
          >
            Search!
          </button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "name": "search",
            "onChange": [Function],
            "placeholder": "Search the Database by any keyword!",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search!",
            "className": "search-btn",
          },
          "ref": null,
          "rendered": "Search!",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <input
            className="search-input"
            name="search"
            onChange={[Function]}
            placeholder="Search the Database by any keyword!"
            type="text"
            value=""
          />
          <button
            className="search-btn"
          >
            Search!
          </button>
        </form>,
        "className": "search-bar-div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="search-input"
              name="search"
              onChange={[Function]}
              placeholder="Search the Database by any keyword!"
              type="text"
              value=""
            />,
            <button
              className="search-btn"
            >
              Search!
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "search-input",
              "name": "search",
              "onChange": [Function],
              "placeholder": "Search the Database by any keyword!",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search!",
              "className": "search-btn",
            },
            "ref": null,
            "rendered": "Search!",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
